File: code/mathematical_algorithms/src/reverse_factorial/reverse_factorial.c
#include <stdio.h>

void
reverse_factorial(long long int factorial)
{
	long long int n = 1,m = 0;
	while (m < 100) {
		m += 1;
		n *= m;
		if (factorial == n) {
			printf("%lli is %lli!\n",factorial,m);
			break;
		}
		else if (m == 100)
			printf("%lli is not a factorial product of any integer\n",factorial);
	}
}

int
main()
{
	long long int factorial;
	printf("Reverse Factorial > ");
	scanf("%lli",&factorial);
	reverse_factorial(factorial);

	return (0);
}

File: code/mathematical_algorithms/src/reverse_factorial/reverse_factorial.py
# Part of Cosmos by OpenGenus Foundation
def reversefactorial(factorial):
    n = 1
    m = 0
    notfound = True
    while m < 100:
        m += 1
        n *= m
        if factorial == n:
            print(str(factorial) + " is " + str(m) + "!")
            notfound = False
            break
        elif m == 100:
            print(str(factorial) + " is not a factorial product of any integer.")


factorial = int(input("Reverse Factorial > "))
reversefactorial(factorial)
